ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -238,6 +240,14 @@ A single compaction job could be performed across multiple compaction cycles (if
   }
 
   /**
+   * @return the number of temporary log segment files this compactor is currently using.
+   */
+  int getSwapSegmentsInUse() {
+    String[] tempSegments = dataDir.list(TEMP_LOG_SEGMENTS_FILTER);
+    return tempSegments == null ? 0 : tempSegments.length;
```

Comment:
`.list()` returning `null` is an error because it means that `dataDir` is not a directory.